dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/SymbolProblem.java

```diff
@@ -38,7 +37,11 @@
   private final ClassFile containingClass;
 
   public SymbolProblem(Symbol symbol, ErrorType errorType, @Nullable ClassFile containingClass) {
-    this.symbol = checkNotNull(symbol);
+    checkNotNull(symbol);
+
+    // After finding symbol problem, there is no need to have SuperClassSymbol over ClassSymbol.
```

Comment:
What does this change do for you? `SuperClassSymbol` is already a subclass of `ClassSymbol`. Why bother removing that information?

(Also, I'm not sure what `SuperClassSymbol` does that's different from `ClassSymbol`. If there's a problem with a reference to a class as a superclass, then that's a kind of `SymbolProblem`, not a different kind of symbol.

Reply:
Because of the inequality (`equals` method) between `ClassSymbol` and `SuperClassSymbol`, the dashboard was showing the symbol problem twice for the same (missing) class; one for normal class symbol and the other for class symbol referenced by subclass.


`SuperClassSymbol` is a marker that the reference is for class inheritance. Once the class inheritance validation ([link](https://github.com/GoogleCloudPlatform/cloud-opensource-java/blob/master/dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/SuperClassSymbol.java#L20)) is done, these symbols should be treated the same.

Reply:
What is the difference between a normal class symbol and a superclass symbol? They're both a class symbol, right? Is the latter just used to report a missing superclass problem? That should be represented as a missing symbol problem, shouldn't it? Just like any other reference to a class.

Reply:
The difference is only to check class inheritance condition when validating reference to `ClassSymbol` including SuperClassSymbol; for example a subclass cannot override `final` methods on superclass.

> That should be represented as a missing symbol problem, shouldn't it?
Yes.